src/nvim/eval.c

```diff
@@ -14456,11 +14456,11 @@ static void f_setreg(typval_T *argvars, typval_T *rettv)
   char_u      *strregname;
   char_u      *stropt;
   bool append = false;
-  char_u yank_type;
+  MotionType yank_type;
   long block_len;
 
   block_len = -1;
-  yank_type = MAUTO;
+  yank_type = kUnknownMotion;
```

Comment:
By the way, why did you choose to rename constants? When I converted VAR_LOCKED/… and VAR_NUMBER/… into enums I preferred to keep old names in order not to fight with lots of linter errors like you will have to now.